vms/platformvm/txs/executor/standard_tx_executor.go

```diff
@@ -940,6 +940,78 @@ func (e *StandardTxExecutor) SetSubnetValidatorWeightTx(tx *txs.SetSubnetValidat
 	return nil
 }
 
+func (e *StandardTxExecutor) IncreaseBalanceTx(tx *txs.IncreaseBalanceTx) error {
+	var (
+		currentTimestamp = e.State.GetTimestamp()
+		upgrades         = e.Backend.Config.UpgradeConfig
+	)
+	if !upgrades.IsEtnaActivated(currentTimestamp) {
+		return errEtnaUpgradeNotActive
+	}
+
+	if err := e.Tx.SyntacticVerify(e.Ctx); err != nil {
+		return err
+	}
+
+	if err := avax.VerifyMemoFieldLength(tx.Memo, true /*=isDurangoActive*/); err != nil {
+		return err
+	}
+
+	// Verify the flowcheck
+	fee, err := e.FeeCalculator.CalculateFee(tx)
+	if err != nil {
+		return err
+	}
+
+	fee, err = safemath.Add(fee, tx.Balance)
+	if err != nil {
+		return err
+	}
+
+	if err := e.Backend.FlowChecker.VerifySpend(
+		tx,
+		e.State,
+		tx.Ins,
+		tx.Outs,
+		e.Tx.Creds,
+		map[ids.ID]uint64{
+			e.Ctx.AVAXAssetID: fee,
+		},
+	); err != nil {
+		return err
+	}
+
+	sov, err := e.State.GetSubnetOnlyValidator(tx.ValidationID)
+	if err != nil {
+		return err
+	}
+
+	// If the validator is currently inactive, we are activating it.
+	if sov.EndAccumulatedFee == 0 {
+		if gas.Gas(e.State.NumActiveSubnetOnlyValidators()) >= e.Backend.Config.ValidatorFeeConfig.Capacity {
+			return errMaxNumActiveValidators
+		}
+
+		sov.EndAccumulatedFee = e.State.GetAccruedFees()
```

Comment:
`sov.EndAccumulatedFee = e.State.GetAccruedFees()` - why do we do this? 

In another PR you do:

`remainingBalance := sov.EndAccumulatedFee - accruedFees` 

Why do you set this here to be equal to it?

Reply:
when the validator is activated it starts paying fees, having a 0 value in EndAccumulatedFee means the validator is inactive, so a fee paying validator (active validator) balance is equal to `sov.EndAccumulatedFee - e.State.GetAccruedFees()`

Reply:
By storing the `EndAccumulatedFee` for each validator how we do, we are able to charge _every_ validator some amount `x` simply by increasing `e.State.GetAccruedFees()` by `x`. Otherwise we would need to iterate over every validator and explicitly deduct their balance.